src/orchestratord/src/controller/materialize/console.rs

```diff
@@ -0,0 +1,187 @@
+const CONSOLE_IMAGE_HTTP_PORT: i32 = 80;
+
+pub struct Resources {
+    console_deployment: Box<Deployment>,
+    console_service: Box<Service>,
+}
+
+impl Resources {
+    pub fn new(config: &super::Args, mz: &Materialize, console_image_ref: &str) -> Self {
+        let console_deployment = Box::new(create_console_deployment_object(
+            config,
+            mz,
+            console_image_ref,
+        ));
+        let console_service = Box::new(create_console_service_object(config, mz));
+        Self {
+            console_deployment,
+            console_service,
+        }
+    }
+
+    pub async fn apply(&self, client: &Client, namespace: &str) -> Result<(), anyhow::Error> {
+        let deployment_api: Api<Deployment> = Api::namespaced(client.clone(), namespace);
+        let service_api: Api<Service> = Api::namespaced(client.clone(), namespace);
+
+        apply_resource(&deployment_api, &self.console_deployment).await?;
+        apply_resource(&service_api, &self.console_service).await?;
+
+        Ok(())
+    }
+}
+
+fn create_console_deployment_object(
+    config: &super::Args,
+    mz: &Materialize,
+    console_image_ref: &str,
+) -> Deployment {
+    let mut pod_template_labels = mz.default_labels();
+    pod_template_labels.insert(
+        "materialize.cloud/name".to_owned(),
+        mz.console_deployment_name(),
+    );
+    pod_template_labels.insert("app".to_owned(), "console".to_string());
+    pod_template_labels.insert(
+        "materialize.cloud/app".to_owned(),
+        mz.console_service_name(),
+    );
+
+    let ports = vec![ContainerPort {
+        container_port: CONSOLE_IMAGE_HTTP_PORT,
+        name: Some("http".into()),
+        protocol: Some("TCP".into()),
+        ..Default::default()
+    }];
+
+    let probe = Probe {
+        http_get: Some(HTTPGetAction {
+            path: Some("/".to_string()),
+            port: IntOrString::Int(CONSOLE_IMAGE_HTTP_PORT),
+            ..Default::default()
+        }),
+        initial_delay_seconds: Some(10),
+        period_seconds: Some(30),
+        ..Default::default()
+    };
+
+    let args = vec![];
+    let env = vec![EnvVar {
+        name: "MZ_ENDPOINT".to_string(),
+        value: Some(format!(
+            "http://{}.{}.svc.cluster.local:{}",
+            mz.environmentd_service_name(),
+            mz.namespace(),
+            config.environmentd_http_port,
+        )),
+        ..Default::default()
+    }];
+
+    if config.enable_tls {
+        unimplemented!();
+    } else {
+        // currently the docker image just doesn't implement tls
+    }
+
+    let container = Container {
+        name: "console".to_owned(),
+        image: Some(console_image_ref.to_string()),
+        image_pull_policy: Some(config.image_pull_policy.to_string()),
+        ports: Some(ports),
+        args: Some(args),
+        env: Some(env),
+        readiness_probe: Some(probe),
+        resources: mz.spec.console_resource_requirements.clone(),
```

Comment:
do we have any defaults on this?

Reply:
no, it'll just default to unconstrained, which seems fine